### ISSUE TITLE
Submit/serialization fails in certain scenario

### ISSUE THREAD

**Issue by kettanaito** [kettanaito/react-advanced-form#219]
## Environment
* **react-advanaced-form:** 1.0.0-alpha

## What
Submitting/serializing the form throws the following error in the Registration from example:
```
Unhandled Rejection (TypeError)
Expected iterable or array-like: undefined
```

## Why
That may be caused by some uncaught exception durning the serialization. May be related to the `flattenDeep` changes introduced with the alpha release.

## How
1. Setup the same scenario in the `examples` folder.
1. Investigate and eliminate the issue.
1. Create an integration test for it / update existing tests.

**Comment by kettanaito** [comment]
## Update
The reason for this issue is missing default argument for `predicate` parameter of `Form.validate()` method. Because it was missing, the found group matched the predicate (which resolves to true in case not provided). That resulted into wrong flattened fields Object upon validation.

The fix is merged to the bugfix branch and is preparing for pull request.

**Comment by kettanaito** [comment]
Merged into `master` and will be published under `1.0`. Closing.
